src/Symfony/Component/Form/FormErrorIterator.php

```diff
@@ -205,7 +205,7 @@ public function hasChildren()
     /**
      * Alias of {@link current()}.
      *
-     * @return self
+     * @return FormError|self
```

Comment:
IMO `self` is correct as logically one should only call `getChildren()` when there are actual child nodes (which in the method above is a check for the current element being an instance of `FormErrorIterator`.

Maybe the method should throw if the current item is not iterable.

Reply:
> Maybe the method should throw if the current item is not iterable.

Makes sense. Should this be considered a BC break or rather a bugfix?

Reply:
I would vote for triggering a deprecation in 5.4 and add the exception in 6.0.

Reply:
Okay, I will open MRs for 5.4 and 6.0 soon